explore/src/main/scala/explore/observationtree/AsterismGroupObsListActions.scala

```diff
@@ -229,42 +236,29 @@ object AsterismGroupObsListActions {
     }
   )
 
-  def targetExistence(
-    targetId: Target.Id
-  )(implicit
-    appCtx:   AppContextIO
-  ) = // can switch appCtx to TransactionalClient when logging not required
-    Action[AsterismGroupsWithObs, Option[TargetGroup]](
-      getter = (agwo: AsterismGroupsWithObs) => agwo.targetGroups.get(targetId),
-      setter = (otg: Option[TargetGroup]) =>
-        (agwo: AsterismGroupsWithObs) =>
-          if (agwo.targetGroups.contains(targetId)) {
-            // Delete - don't need to worry about the asterism groups for delete
-            val newGroups = agwo.targetGroups.removed(targetId)
-            agwo.copy(targetGroups = newGroups)
-          } else
-            // Undelete: Should always have a TargetGroup at this point.
-            otg.fold(agwo) { tg =>
-              val newTargetGroups   = agwo.targetGroups.updated(targetId, tg)
-              val obsInTg           = tg.obsIds
-              // update the asterism groups
-              val newAsterismGroups = agwo.asterismGroups
-                .map { case (_, ag) =>
-                  if (ag.obsIds.toSortedSet.intersect(obsInTg).nonEmpty) ag.addTargetId(targetId)
-                  else ag
-                }
-                .toList
-                .toSortedMap(_.obsIds)
-              agwo.copy(targetGroups = newTargetGroups, asterismGroups = newAsterismGroups)
-            }
+  def targetExistence(targetId: Target.Id, setPage: Option[Target.Id] => IO[Unit])(implicit
+    c:                          TransactionalClient[IO, ObservationDB]
+  ): Action[AsterismGroupsWithObs, Option[TargetWithObs]] =
+    // can switch appCtx to TransactionalClient when logging not required
+    Action[AsterismGroupsWithObs, Option[TargetWithObs]](
+      getter = (agwo: AsterismGroupsWithObs) => agwo.targetsWithObs.get(targetId),
+      setter = (otwo: Option[TargetWithObs]) =>
```

Comment:
Oi. I can't understand this stuff, anymore. 🙄 I'll have to trust that you tested it...

Reply:
It took me a bit to remember how to use this.

Basically, these are instructions on how to get/set a `Target` with a certain `targetId` in an `AsterismGroupsWithObs` (which is the structure that holds asterisms, targets and observations).

Passing/getting `None` means deletion/absence, wheareas `Some` means adding/existence.

When a target is deleted or added with associated observations, the asterisms of those observations have to be adjusted.